packages/squiggle-lang/src/rescript/FunctionRegistry/FunctionRegistry_Core.res

```diff
@@ -91,104 +69,43 @@ module FRType = {
     | FRTypeAny => `any`
     }
 
-  let rec toFrValue = (r: Reducer_T.value): option<frValue> =>
-    // not all value variants are supported, but it's not important (we'll probably deprecate frValues soon anyway)
-    switch r {
-    | IEvNumber(f) => Some(FRValueNumber(f))
-    | IEvString(f) => Some(FRValueString(f))
-    | IEvDistribution(f) => Some(FRValueDistOrNumber(FRValueDist(f)))
-    | IEvLambda(f) => Some(FRValueLambda(f))
-    | IEvArray(elements) =>
-      elements->E.A2.fmap(toFrValue)->E.A.O.openIfAllSome->E.O2.fmap(r => FRValueArray(r))
-    | IEvRecord(map) =>
-      Belt.Map.String.toArray(map)
-      ->E.A2.fmap(((key, item)) => item->toFrValue->E.O2.fmap(o => (key, o)))
-      ->E.A.O.openIfAllSome
-      ->E.O2.fmap(r => FRValueRecord(r))
-    | _ => None
-    }
-
-  let rec matchWithExpressionValue = (t: t, r: Reducer_T.value): option<frValue> =>
+  let rec matchWithValue = (t: t, r: Reducer_T.value): bool =>
```

Comment:
FYI, I really didn't like this ``match`` code, imagine it could be replaced/refactored sometime.

Reply:
Yeah, I expect it'll evolve over time as we get more type-checking capabilities in the interpreter.

Before this PR, we had 3 extra checks/conversions for each function call (this match; fr value conversion; switches in the definition). Now it's down to 2.

I hope it'll go to 1 with smarter FunctionRegistry APIs, and then to 0 on non-polymorphic functions when we get to the strong typing.

`frType` should be unified with `internalExpressionValueType`, but that's too much work for this PR (and I haven't still reviewed Umur's type checking work carefully yet).